core/src/main/java/io/fabric8/maven/core/service/openshift/ImageStreamService.java

```diff
@@ -205,29 +208,59 @@ private String findTagSha(OpenShiftClient client, String imageStreamName, String
             if (tags == null || tags.isEmpty()) {
                 continue;
             }
-            // latest tag is the first
+
+            // Iterate overall tags and get the latest one by 'created' attribute
+            TagEvent latestTag = null;
+
             TAG_EVENT_LIST:
             for (NamedTagEventList list : tags) {
                 List<TagEvent> items = list.getItems();
-                if (items == null) {
+                if (items == null || items.isEmpty()) {
                     continue TAG_EVENT_LIST;
                 }
 
-                // latest item is the first
-                for (TagEvent item : items) {
-                    String image = item.getImage();
-                    if (Strings.isNotBlank(image)) {
-                        log.info("Found tag on ImageStream " + imageStreamName + " tag: " + image);
-                        return image;
-                    }
+                for (TagEvent tag : items) {
+                    latestTag = latestTag == null ? tag : newerTag(tag, latestTag);
                 }
             }
+
+            if (latestTag != null && Strings.isNotBlank(latestTag.getImage())) {
+                String image = latestTag.getImage();
+                log.info("Found tag on ImageStream " + imageStreamName + " tag: " + image);
+                return image;
+            }
         }
+
         // No image found, even after several retries:
         if (currentImageStream == null) {
             throw new MojoExecutionException("Could not find a current ImageStream with name " + imageStreamName + " in namespace " + namespace);
         } else {
             throw new MojoExecutionException("Could not find a tag in the ImageStream " + imageStreamName);
         }
     }
+
+    public TagEvent newerTag(TagEvent tag1, TagEvent tag2) {
+        Date tag1Date = extractDate(tag1);
+        Date tag2Date = extractDate(tag2);
+
+        if(tag1Date == null) {
+            return tag2;
+        }
+
+        if(tag2Date == null) {
+            return tag1;
+        }
+
+        return tag1Date.compareTo(tag2Date) > 0 ? tag1 : tag2;
+    }
+
+    private Date extractDate(TagEvent tag) {
```

Comment:
There is an almost similar function [KubernetesResourceUtil.parseDate](https://github.com/fabric8io/fabric8-maven-plugin/blob/master/core/src/main/java/io/fabric8/maven/core/util/KubernetesResourceUtil.java#L622); I think we should use that instead of this; via minor changes of course ;)

Reply:
@rohanKanojia good suggestion. Just wanted to handle `Null` check and string formate is different.